rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -2110,6 +2110,148 @@ private static Stream<BridgeConstants> provideBridgeConstants() {
+        when(provider.getNewFederationBtcUTXOs())
+            .thenReturn(utxosNew);
+
+        // Advance blockchain to migration phase
+        BlockGenerator blockGenerator = new BlockGenerator();
+        org.ethereum.core.Block rskCurrentBlock = blockGenerator.createBlock(180, 1);
+        BridgeEventLogger bridgeEventLogger = mock(BridgeEventLogger.class);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstants)
+            .withProvider(provider)
+            .withEventLogger(bridgeEventLogger)
+            .withExecutionBlock(rskCurrentBlock)
+            .withActivations(activations)
+            .build();
+
+        Transaction tx = Transaction
+            .builder()
+            .nonce(NONCE)
+            .gasPrice(GAS_PRICE)
+            .gasLimit(GAS_LIMIT)
+            .destination(Hex.decode(TO_ADDRESS))
+            .data(Hex.decode(DATA))
+            .chainId(Constants.REGTEST_CHAIN_ID)
+            .value(DUST_AMOUNT)
+            .build();
+        bridgeSupport.updateCollections(tx);
+
+
+        // Assert two pegouts are added to pegouts waitingForConfirmations, one pegout batch and one pegout migration tx
+        assertEquals(2, provider.getPegoutsWaitingForConfirmations().getEntries().size());
+        // Assert the two added pegouts has the same pegoutCreationRskTxHash
+        assertEquals(1,
+            provider
+                .getPegoutsWaitingForConfirmations()
+                .getEntries()
+                .stream()
+                .map(PegoutsWaitingForConfirmations.Entry::getPegoutCreationRskTxHash)
+                .collect(Collectors.toSet()).size()
+        );
+
+        // Advance blockchain to the height where both pegouts have enough confirmations to be moved to waitingForSignature
+        rskCurrentBlock = blockGenerator.createBlock(184, 1);
+        bridgeSupport = bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstants)
+            .withProvider(provider)
+            .withEventLogger(bridgeEventLogger)
+            .withExecutionBlock(rskCurrentBlock)
+            .withActivations(activations)
+            .build();
+
+        tx = Transaction
+            .builder()
+            .nonce(NONCE)
+            .gasPrice(GAS_PRICE)
+            .gasLimit(GAS_LIMIT)
+            .destination(Hex.decode(TO_ADDRESS))
+            .data(Hex.decode(DATA))
+            .chainId(Constants.REGTEST_CHAIN_ID)
+            .value(DUST_AMOUNT)
+            .build();
+        bridgeSupport.updateCollections(tx);
+
+        // Since only one pegout per updateCollection call is move to waitingForisgnatures
+        // -assert one of the two pegout get moved to pegoutWaitingForSignate and one is left on waitingForConfirmation
+        assertEquals(1, provider.getPegoutsWaitingForConfirmations().getEntries().size());
+        assertEquals(1, provider.getPegoutsWaitingForSignatures().size());
```

Comment:
Is there are a way to distinguish which one is the migration and which one is the pegout? Because it'd be really interesting to assert that the migration transactions is confirmed first

Reply:
Done